docker_cli/_image.py

```diff
@@ -0,0 +1,471 @@
+            bash installed.
+
+        Parameters
+        ----------
+        cmd : str
+            The desired command, in linux command format.
+        stdout : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        stderr : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        interactive : bool, optional
+            A boolean describing whether or not to run this command in
+            interactive mode or not. Defaults to True.
+        network : str, optional
+            The name of the network. Defaults to "host".
+
+        Returns
+        -------
+        str, optional
+            The output of the process, if `stdout` == PIPE.
+
+        Raises
+        -------
+        CommandNotFoundOnImageError:
+            When a command is attempted that is not recognized on the image.
+        """
+        cmd = ["docker", "run", f"--network={network}", "--rm"]
+        if interactive:
+            cmd += ["-i"]
+            if stdin.isatty():
+                cmd += ["--tty"]
+        cmd += [self._id, "bash"]
+        cmd += ["-ci"] if interactive else ["-c"]
+        cmd += split(f"'{command}'")
+
+        try:
+            result = run(
+                cmd,
+                text=True,
+                stdout=stdout,  # type: ignore
+                stderr=stderr,  # type: ignore
+                check=True
+            )
+            retval = result.stdout
+        except CalledProcessError as err:
+            if err.returncode == 127:
+                raise CommandNotFoundError(split(command)[0]) from err
+            else:
+                raise
+        return retval
+
+    def drop_in(
+            self,
+            network: str = "host"
+    ):
+        """
+        Start a drop-in session on a disposable container.
+
+        .. warning::
+            This method does not work correctly if the image built does not have
+            bash installed.
+
+        Parameters
+        ----------
+        network : str, optional
+            The name of the network. Defaults to "host".
+
+        Raises
+        -------
+        CommandNotFoundOnImageError:
+            When bash is not recognized on the image.
+        """
+        cmd = ["docker", "run", f"--network={network}", "--rm", "-i"]
+        if stdin.isatty():
+            cmd += ["--tty"]
+        cmd += [self._id, "bash"]
+
+        try:
+            run(
+                cmd,
+                text=True,
+                check=True
+            )
+        except CalledProcessError as err:
+            if err.returncode == 127:
+                raise CommandNotFoundError("bash") from err
+            else:
+                print(f"Drop-in session exited with code {err.returncode}.")
+                return
+        print("Drop-in session exited with code 0.")
```

Comment:
This should just be implemented in terms of `Image.run()`.

```suggestion
        self.run("bash", interactive=True, network=network)
```

Reply:
This runs `docker run --network=host --rm -i --tty {self.id} bash -c bash` as written - would it be best to add a special case under `image.run()` for when "bash" is passed in so it just runs `bash` instead of `bash -c bash`?

Reply:
Nah, I don't think it's necessary to have a special case for "bash". It's not really an issue to be running nested shell sessions like this. Technically, the default [entrypoint](https://docs.docker.com/engine/reference/builder/#entrypoint) for most of these images is probably a bash shell as well, so we're really running bash inside of bash inside of bash.

If we want to avoid this in the future, I think a more robust solution would be to allow the user to specify a custom entrypoint (via the `--entrypoint` flag of [`docker run`](https://docs.docker.com/engine/reference/commandline/run/)) in addition to the command, rather than adding a special case for "bash". That way, the user would have more control over exactly what `docker run` executes -- they could make "bash" the `entrypoint` and leave the `command` empty in order to drop into a bash shell without any nesting. But let's not worry about this for the time being.